agent/agent.py

```diff
@@ -257,26 +259,30 @@ class send_file:
     """Wrapper that represents Flask.send_file functionality."""
 
     def __init__(self, path, encoding):
+        self.length = None
         self.path = path
         self.status_code = 200
         self.encoding = encoding
 
+    def okay_to_send(self):
+        return os.path.isfile(self.path) and os.access(self.path, os.R_OK)
+
     def init(self):
-        if os.path.isfile(self.path) and os.access(self.path, os.R_OK):
-            self.length = os.path.getsize(self.path)
+        if self.okay_to_send():
+            if self.encoding != BASE_64_ENCODING:
+                self.length = os.path.getsize(self.path)
```

Comment:
This is the fix.  Previously we were always setting the length.  Now, we'll skip setting the length when base64 transmission is called for.